lib/rubygems/request_set/gem_dependency_api.rb

```diff
@@ -526,6 +552,7 @@ def gem_requires name, options # :nodoc:
     else
       @requires[name] << name
     end
+    raise "Unhandled gem options #{options.inspect}" unless options.empty?
```

Comment:
Turns out I was using the system version of rubygems instead of the git repo.  Can you please show how to run git repo?  I tried `../rubygems/bin/gem install --file` from the haystack project